web/static/js/components/ui/Dropdown.jsx

```diff
@@ -50,6 +52,44 @@ DropdownItem.propTypes = {
   className: PropTypes.string
 }
 
+export class DropdownDropdownCheckboxItem extends Component {
+  static propTypes = {
+    onClick: PropTypes.func,
+    defaultChecked: PropTypes.bool,
+    id: PropTypes.string,
+    displayText: PropTypes.string
+  }
+
+  componentDidMount() {
+    const { onClick } = this.props
+    $(this.refs.li).on('click', function(event) {
+      // Prevent the menu from closing
+      event.stopPropagation()
+    })
+    $(this.refs.input).on('click', function(event) {
+      // Prevent the menu from closing
+      event.preventDefault()
+      onClick()
+    })
+  }
+
+  render() {
+    const { defaultChecked, id, displayText } = this.props
+    return (
+      <li ref='li' className='checkbox'>
+        <input
+          id={id}
+          ref='input'
+          type='checkbox'
+          defaultChecked={defaultChecked}
+          className='filled-in'
+        />
+        <label htmlFor={id}>{displayText}</label>
```

Comment:
if the label is the paren of the input, there will be no need to use htmlFor.

If that can't be done, see https://www.npmjs.com/package/react-html-id for how to generate ids locally

Reply:
It seems to me that placing the input as a child of the label complicates styling. Because [_checkboxes.scss](https://github.com/instedd/surveda/blob/74cfc581a131719caf36879894a8f8f0b075e60d/web/static/vendor/css/materialize/components/forms/_checkboxes.scss) assumes that the label is after or before the checkbox.

Regarding the `react-html-id`, Surveda doesn't use it yet. Do you want to start using it now?

For the moment, I generated the id using `lodash.uniqueId`. I know it's not the same because the scope of the uniqueness is different. But we're already using it for similar purposes [here](https://github.com/instedd/surveda/blob/5bc348047de52eaab70397a20ebb747baaac3a10/web/static/js/components/ui/Dropdown.jsx#L18).

I propose opening a technical debt issue to start using `react-html-id` in the whole application.

Reply:
Using `lodash.uniqueId` is fine, but the value returned from it should be computed during component creation or mount and stored it in component state. Otherwise, each render will keep changing the id.